aiosmtpd/smtp.py

```diff
@@ -436,10 +458,10 @@ async def _handle_client(self):
                     # send error response and read the next command line.
                     await self.push('500 Command line too long')
                     continue
-                log.debug('_handle_client readline: %r', line)
+                log.debug('_handle_client readline: %r', sanitize(line))
                 # XXX this rstrip may not completely preserve old behavior.
                 line = line.rstrip(b'\r\n')
-                log.info('%r >> %r', self.session.peer, line)
+                log.info('%r >> %r', self.session.peer, sanitize(line))
```

Comment:
**suggestion**: Would it make more sense to create a `sanitized_log_debug`/`info`, rather than depend on remembering to call `sanitize`?

Reply:
Well, that would only change the need of remembering to call `sanitize` with the need to call `sanitized_log_*`, so I don't really see the benefits.

Reply:
It's definitely a small thing, but it boils down to this: 

```
log.debug('blahahahahahblaksj %s', whatever)
log.debug('bargbargbargbarg arg %s', sanitize (bob))
```

Is easy to miss that one has the sanitize call and the other is missing. Especially in a huge block of code.

It's even worse with a multitude of parameters.

Is it the end of the world? No. But are there tangible benefits that make writing and reviewing easier? Sure 😁

Basically any time you see a logging call that sends sensitive data, it should be calling the sensitive log function.

So... A small thing that I don't think should block the PR. But it would be a useful thing 🙂

Reply:
Hmmm, you do have a point there.

Let me mull it over... I definitely don't want to make a too-complex sanitizing system... but I think I have some inkling on how to implement it nicely...

Reply:
I think I was actually reviewing this first on mobile :upside_down_face: 

I'm OK if we want to merge this first and then do that in another PR. Otherwise, the thought I had during discussion is that we could either:

a) create a logger/handler implementation, so it's literally created like:

```
sanitized_logger = CustomLogger()
...
sanitized_logger.debug(...)
```

b) create a function/group of functions that live somewhere, so

```
from utils import sanitized_debug
...
sanitized_debug(...)
```

or similar.

Reply:
I much prefer option (b), it's less complex and can be implemented cleanly.

Rather than wrapping all calls, I think of going like this:

```python
def sanitized_log(func, msg, *args):
    msg = sanitize(msg)
    gerard(log, func)(msg, *args)
```

Where sanitizing is needed, we change the call

```
log.debug(...)
```

to

```
sanitized_log("debug", ...)
```

What do you think?

Reply:
Maybe just `sanitized_log(log.debug, ...)` then skip the getattr call? Otherwise I'm +1 for that approach

Reply:
That's also good!

I'll implement that asap.